Add  maintenance time section in  appendix-07-nightly-rust.md (#3859)
rust-lang/book: src/appendix-07-nightly-rust.md

next stable release happens! Breakage in a beta release is relatively rare, but
`rustc` is still a piece of software, and bugs do exist.

### Maintenance time

> The community only supports the most recent stable version, which means the 

Reviewer: I would say "The Rust project", not "The community" here

Author: OK, I was wondering whether to use "The Rust project" or "The Rust" or "The community", all of them are fine to me

> The Rust project only supports the most recent stable version, which means the 
> maintenance time of a stable version is six weeks. When a new stable version 
> released, the old version reaches its EOL(end-of-life). 

Reviewer 2: I like this addition but tweaked the wording a bit.

```suggestion
The Rust project supports the most recent stable version. When a new stable
version is released, the old version reaches its end of life (EOL). This means
each version is supported for six weeks.
```
